angular/projects/admin/src/app/services/document-type.service.ts

```diff
@@ -31,11 +31,11 @@ export class DocumentTypeService {
       .valueChanges();
   }
 
-  async create(id: string) {
-    const docRef = this.siteCollection.collection('document-types').doc(id);
+  async create(title: string, slug: string) {
```

Comment:
Try to not expand the object's field attributes into a list of parameters. Service APIs should as much as possible take either 

 1. The "object type" as a parameter
 2. Just a string `id`

E.g. either

```typescript
create('some-id')
```

or

```typescript
const documentType = {...} as DocumentType;
create(documentType)
```
If you need a function that does something in particular, then you make a function called `createWithTitle` that only takes the title and slugification is inside of the service.

Reply:
So change the name of `create` to `createWithTitle` and the only parameter to the function is the **non slugified** title. Move `_slugify` inside the service.

Reply:
Resolve on this commit https://github.com/oddbit/tanam/pull/59/commits/8fca9d95aa49ab745896b6bf3d23e3cd48589233